test/Handler/CommentsSpec.hs

```diff
@@ -11,30 +17,32 @@ spec :: Spec
 spec = withApp $ do
     describe "GET CommentsR" $ do
         it "returns a list of comments by article" $ do
-            Entity siteId _ <- createSite
-
-            u <- createUser "1"
-            c1 <- createComment (entityKey u) siteId "1" "1" "1"
-            c2 <- createComment (entityKey u) siteId "1" "2" "2"
-            c3 <- createComment (entityKey u) siteId "2" "1" "3"
+            siteId <- runDB $ insert $ buildSite
+            user <- runDB $ createUser "1"
+            (c1, c2, c3) <- runDB $ (,,)
```

Comment:
Yes, their types changed from `YesodExample App a` to `DB a`. They previously called `runDB` internally and no longer do. This has a few benefits:

- `runDB` can live separately from `Factories` without a cyclic import
- multiple factory functions can be composed into a single `runDB` call (there are cases where the Applicative cleverness is not needed to achieve this)

Eventually, I want to take this further and have clear separation between `runDB`, `insert` and `buildFoo` within any `createFoo` functions. This is the only way we can get rid of the multiple arguments to `createComment` which I've always found obfuscating.

For example,

```hs
-- we often only care about Thread, but we have to pass everything always
createComment "1" "1" "1"

-- I'd like this kind of composability instead
runDB $ insert $ buildComment { commentThread "1" }
```

`createComment` could still exist as `runDB $ insert buildComment` for cases where you don't care about the comment data itself.